homeassistant/components/switch/vera.py

```diff
@@ -0,0 +1,169 @@
+"""
+Support for Vera switches.
+
+Configuration:
+To use the Vera lights you will need to add something like the following to
+your config/configuration.yaml
+
+switch:
+    platform: vera
+    vera_controller_url: http://YOUR_VERA_IP:3480/
+    device_data:
+        12:
+            name: My awesome switch
+            exclude: true
+        13:
+            name: Another Switch
+
+VARIABLES:
+
+vera_controller_url
+*Required
+This is the base URL of your vera controller including the port number if not
+running on 80
+Example: http://192.168.1.21:3480/
+
+
+device_data
+*Optional
+This contains an array additional device info for your Vera devices.  It is not
+required and if not specified all lights configured in your Vera controller
+will be added with default values.  You should use the id of your vera device
+as the key for the device within device_data
+
+
+These are the variables for the device_data array:
+
+
+name
+*Optional
+This parameter allows you to override the name of your Vera device in the HA
+interface, if not specified the value configured for the device in your Vera
+will be used
+
+
+exclude
+*Optional
+This parameter allows you to exclude the specified device from homeassistant,
+it should be set to "true" if you want this device excluded
+
+"""
+import logging
+import time
+from requests.exceptions import RequestException
+
+from homeassistant.helpers import ToggleDevice
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL, ATTR_TRIPPED, ATTR_ARMED, ATTR_LAST_TRIP_TIME)
+# pylint: disable=no-name-in-module, import-error
+import homeassistant.external.vera.vera as veraApi
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def get_devices(hass, config):
+    """ Find and return Vera switches. """
+
+    base_url = config.get('vera_controller_url')
+    if not base_url:
+        _LOGGER.error(
+            "The required parameter 'vera_controller_url'"
+            " was not found in config"
+        )
+        return False
+
+    device_data = config.get('device_data', {})
+
+    vera_controller = veraApi.VeraController(base_url)
+    devices = []
+    try:
+        devices = vera_controller.get_devices(['Switch', 'Armable Sensor'])
+    except RequestException as inst:
+        # There was a network related error connecting to the vera controller
+        _LOGGER.error("Could not find Vera switches: %s", inst)
+        return False
+
+    vera_switches = []
+    for device in devices:
+        extra_data = device_data.get(device.deviceId, None)
```

Comment:
If you replace `None` here with `{}` you will get back an empty hash. That way you can do `exclude = extra_data.get('exclude', False)` directly and omit `if extra_data`.